### ISSUE TITLE
Mobile Client: Back button not working once you try to search for a deployment

### ISSUE THREAD

**Issue by tuxpiper** [ushahidi/platform#4703]
**Describe the bug**
Back button not working once you try to search for a deployment

**To Reproduce**

1. Open mzima mobile client in xcode or android studio
2. Go to profile > switch deployment, you will find the deployments screen where there is a search input and list of deployments to choose from
3. Click on the search input to open up search in full screen
4. Click on the back button to try to go back to the deployments screen
5. See that the back button does not work

**Expected behavior**
Back button should take user back to the deployments screen

**Screenshots**

![50292ef3-4850-4822-895d-f0bf5b7b1109 (3)](https://github.com/ushahidi/platform/assets/8624777/f26333c4-a316-4eb2-b5b0-ff8c87d54be9)

---
Created automatically from USH-927 (https://linear.app/ushahidi/issue/USH-927/mobile-client-back-button-not-working-once-you-try-to-search-for-a)

**Comment by Remi-dee** [comment]
Hi @Angamanga, I’m an Outreachy Applicant and  I’m happy to inform you that I’ve successfully setup my development environment for mobile client and I’d love to work on this issue,

**Comment by amooesther** [comment]
hello @Angamanga My name is Esther i am an outreachy applicant, please i have setup my developer environment can i will love to work on this project

**Comment by Angamanga** [comment]
@Remi-dee Well done! I'm assigning you to this issue 👍

**Comment by Remi-dee** [comment]
> @Remi-dee Well done! I'm assigning you to this issue 👍

Hi @Angamanga, I'd love to mention that i was unable to reproduce this behavior as the app goes back to the deployment screen (as expected) when back button is clicked(see gif below).

However, i'd love to seek clarification if this was because i had only run the mobile client on browser and not on android studio/xcode? 

Do i neccessarily need to use an android studio/xcode to reproduce this behavior?


![outreachy](https://github.com/ushahidi/platform/assets/96704300/d4d2178e-33e2-450f-b263-599becb81f05)

**Comment by Angamanga** [comment]
@Remi-dee This might be the case, sometimes issues aren't reproducible. I you would like to verify the issue it would be of great help to me, please download our app on app-store or google play and try reproducing it from there, and then let me know in a comment to this thread. Thank you very much 🙏

**Comment by Remi-dee** [comment]
Alright, i'd try this out and let you know what i've discovered.

**Comment by Remi-dee** [comment]
Hello @Angamanga trust you're well, so after carefully testing  Ushahidi Mobile App from Appstore on IOS, i am still unable to reproduce this behaviour as the back button works as it should be: (see gif below).

However, i have not been able to replicate this test on an android mobile device since i don't own one but i'll do that before the end of today and update with this thread so we can verify this is the same behavior on all platform.


https://github.com/ushahidi/platform/assets/96704300/40c4bda1-d26d-4b92-9949-8a4140074f42

**Comment by Angamanga** [comment]
@Remi-dee Thank you for retesting and verifying the issue has been fixed 🙏
